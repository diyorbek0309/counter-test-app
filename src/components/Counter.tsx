import { useState } from "react";

interface CounterProps {
  title: string;
}

const Counter = ({ title }: CounterProps) => {
  const [count, setCount] = useState(0);
  const handleClick = () => {
    setCount(count + 1);
  };

  return (
    <div>
      <h1 data-testid="header">{title}</h1>
      <h2 data-testid="counter">{count}</h2>
      <button type="button" onClick={handleClick} data-testid="button">
        Click
      </button>
    </div>
  );
};

export default Counter;
