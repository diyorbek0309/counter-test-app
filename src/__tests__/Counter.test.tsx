import Counter from "../components/Counter";
import { render } from "@testing-library/react";

describe("This suit is to test the Counter component", () => {
  test("Snapshot of Counter", () => {
    const { asFragment } = render(<Counter title="hello world" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
