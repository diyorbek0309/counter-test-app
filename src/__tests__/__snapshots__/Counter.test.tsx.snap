// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component Snapshot of Body 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="header"
    >
      hello world
    </h1>
    <h2
      data-testid="counter"
    >
      0
    </h2>
    <button
      data-testid="button"
      type="button"
    >
      Click
    </button>
  </div>
</DocumentFragment>
`;

exports[`This suit is to test the Counter component Snapshot of Body 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="header"
    >
      hello world
    </h1>
    <h2
      data-testid="counter"
    >
      0
    </h2>
    <button
      data-testid="button"
      type="button"
    >
      Click
    </button>
  </div>
</DocumentFragment>
`;

exports[`This suit is to test the Counter component Snapshot of Counter 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="header"
    >
      hello world
    </h1>
    <h2
      data-testid="counter"
    >
      0
    </h2>
    <button
      data-testid="button"
      type="button"
    >
      Click
    </button>
  </div>
</DocumentFragment>
`;
